src/core/mac/mac.cpp

```diff
@@ -264,7 +264,17 @@ void Mac::StartEnergyScan(void)
 
 extern "C" void otPlatRadioEnergyScanDone(otInstance *aInstance, int8_t aEnergyScanMaxRssi)
 {
-    aInstance->mThreadNetif.GetMac().EnergyScanDone(aEnergyScanMaxRssi);
+    if (aInstance->mLinkRawEnabled)
+    {
+        if (aInstance->mLinkRawEnergyScanDoneCallback)
+        {
+            aInstance->mLinkRawEnergyScanDoneCallback(aInstance, aEnergyScanMaxRssi);
```

Comment:
The receive callback definitely shouldn't NULL, but the others could be, if that's what everyone wants.